ZOOKEEPER-4862. Add doc to secure prometheus support (#2190)
apache/zookeeper: zookeeper-docs/src/main/resources/markdown/zookeeperMonitor.md

<a name="Prometheus"></a>

> ### Pre-requisites:
> - Enable the `Prometheus MetricsProvider` by setting the following in `zoo.cfg`:
>   ```bash
>   metricsProvider.className=org.apache.zookeeper.metrics.prometheus.PrometheusMetricsProvider
>   ```

Reviewer: ````suggestion
### Pre-requisites:

- Enable the Prometheus MetricsProvider by setting the following in `zoo.cfg`:

  ```conf 
 
 metricsProvider.className=org.apache.zookeeper.metrics.prometheus.PrometheusMetricsProvider
  ```
````

> - The port for Prometheus metrics can be configured using:
>   ```bash
>   metricsProvider.httpPort=7000  # Default port is 7000
>   ```

Reviewer: ````suggestion
- The port for Prometheus metrics can be configured using (default to 7000):
  ```conf
  metricsProvider.httpPort=7000
  ```
````

> #### Enabling HTTPS for Prometheus Metrics:
> ZooKeeper also supports SSL for Prometheus metrics, which provides secure data transmission. To enable this, configure an HTTPS port and set up SSL certificates as follows:

Reviewer: ```suggestion
#### Enabling HTTPS for Prometheus Metrics:

ZooKeeper supports SSL for Prometheus metrics, which provides secure data transmission. To enable this, configure an HTTPS port and set up SSL certificates as follows:
```

- Define the HTTPS port:

>   ```bash

Reviewer: ````suggestion
  ```conf
````

  metricsProvider.httpPorts=4443
  ```

- Configure the SSL key store (holds the server’s private key and certificates):

>   ```bash

Reviewer: ```suggestion
  ```conf
```

  metricsProvider.ssl.keyStore.location=/path/to/keystore.jks
  metricsProvider.ssl.keyStore.password=your_keystore_password
  metricsProvider.ssl.keyStore.type=jks  # Default is JKS
  ```

- Configure the SSL trust store (used to verify client certificates):

>   ```bash

Reviewer: ```suggestion
  ```conf
```

  metricsProvider.ssl.trustStore.location=/path/to/truststore.jks
  metricsProvider.ssl.trustStore.password=your_truststore_password
  metricsProvider.ssl.trustStore.type=jks  # Default is JKS
  ```

> - **Note**: You can enable both HTTP and HTTPS simultaneously by defining both ports:
>   ```bash
>   metricsProvider.httpPorts=7000
>   metricsProvider.httpsPorts=4443
>   ```

Reviewer: ````suggestion
- You can enable HTTP and HTTPS simultaneously by defining both ports:
  ```conf
  metricsProvider.httpPorts=7000
  metricsProvider.httpsPorts=4443
  ```
````

<a name="Prometheus"></a>

### Pre-requisites:
- Enable the `Prometheus MetricsProvider` by setting the following in `zoo.cfg`:
  ```conf
  metricsProvider.className=org.apache.zookeeper.metrics.prometheus.PrometheusMetricsProvider
  ```

- The port for Prometheus metrics can be configured using:
  ```conf
  metricsProvider.httpPort=7000  # Default port is 7000
  ```

#### Enabling HTTPS for Prometheus Metrics:

ZooKeeper also supports SSL for Prometheus metrics, which provides secure data transmission. To enable this, configure an HTTPS port and set up SSL certificates as follows:

- Define the HTTPS port:
  ```conf

>   metricsProvider.httpPorts=4443

Reviewer 2: ```suggestion
  metricsProvider.httpsPort=4443
```

  ```

- Configure the SSL key store (holds the server’s private key and certificates):
  ```conf
  metricsProvider.ssl.keyStore.location=/path/to/keystore.jks
  metricsProvider.ssl.keyStore.password=your_keystore_password

>   metricsProvider.ssl.keyStore.type=jks  # Default is JKS

Reviewer 2: backlog: I think we cloud detect this by filename just as `zookeeper.ssl.keyStore.type`.

Author: Yes, if needed, we can do it.

  ```

- Configure the SSL trust store (used to verify client certificates):
  ```conf
  metricsProvider.ssl.trustStore.location=/path/to/truststore.jks
  metricsProvider.ssl.trustStore.password=your_truststore_password
  metricsProvider.ssl.trustStore.type=jks  # Default is JKS
  ```

- **Note**: You can enable both HTTP and HTTPS simultaneously by defining both ports:
  ```conf

>   metricsProvider.httpPorts=7000
>   metricsProvider.httpsPorts=4443

Reviewer 2: ```suggestion
  metricsProvider.httpPort=7000
  metricsProvider.httpsPort=4443
```

Author: Thanks for catching that, @kezhuw. That was a simple oversight on my part.
